pkg/receive/writer.go

```diff
@@ -70,6 +75,25 @@ func (r *Writer) Write(ctx context.Context, tenantID string, wreq *prompb.WriteR
 		errs errutil.MultiError
 	)
 	for _, t := range wreq.Timeseries {
+		// Check if time series labels are valid. If not, skip the time series
+		// and report the error.
+		err := labelpb.ValidateLabels(t.Labels)
+		if err != nil {
+			switch err {
+			case labelpb.ErrOutOfOrderLabels:
+				numLabelsOutOfOrder++
+				level.Debug(tLogger).Log("msg", "Out of order labels in the label set", "lset", t.Labels)
+			case labelpb.ErrDuplicateLabels:
+				numLabelsDuplicates++
+				level.Debug(tLogger).Log("msg", "Duplicate labels in the label set", "lset", t.Labels)
+			case labelpb.ErrEmptyLabels:
+				numLabelsEmpty++
+				level.Debug(tLogger).Log("msg", "Labels with empty name in the label set", "lset", t.Labels)
+			}
+
+			continue
```

Comment:
Should we add any default case just in case someone modify verifier and forgets to update here? (:

Reply:
Good idea :+1: